pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -879,7 +905,14 @@ func (oe *operationExecutor) generateUnmountDeviceFunc(
 				deviceToDetach.VolumeSpec.Name(),
 				err)
 		}
-
+		refs, err := attachableVolumePlugin.GetDeviceMountRefs(deviceMountPath)
```

Comment:
Yes, I think this check is needed. During kubelet restart experiments, I have seen the system may in a state that reconcile tries to unmount device but there are still some mounting references exist for the device. This check will prevent unmount device before all the mounting references are cleaned up.